server/interpreter.js

```diff
@@ -171,23 +174,55 @@ Interpreter.prototype.createThread = function(runnable) {
     runnable = new Interpreter.State(runnable, this.global);
   }
   var id = this.threads.length;
-  var thread = new Interpreter.Thread(id, runnable, this.now());
+  var thread = new Interpreter.Thread(id, runnable, runAt || this.now());
   this.threads.push(thread);
+  if (this.running) {
+    this.start();
+  }
   return id;
 };
 
 /**
- * Schedule the next runnable thread.  Return true if successful;
- * false otherwise.  If there are no remaining (non-ZOMBIE) threads,
- * also set .done to true.
- * @return {boolean} True iff there is a runnable thread.
+ * Create a new thread to execute a particular function call.
+ * @param {!Interpreter.prototype.Function} fun Function to call.
+ * @param {Interpreter.Value} funcThis value of 'this' in function call.
+ * @param {!Array<Interpreter.Value>} args Arguments to pass.
+ * @param {number=} runAt Time at which thread should begin execution
+ *     (defaults to now).
+ * @return {number} thread ID.
+ */
+Interpreter.prototype.createThreadForFuncall =
```

Comment:
In the world of functional programming the spelling is invariably "funcall" (no double C), after the lisp keyword of the same name.  But FuncCall will probably be clearer here.